neofs_contract.go

```diff
@@ -369,6 +381,91 @@ func Main(op string, args []interface{}) interface{} {
 		}
 
 		return false
+	case "Config":
+		if len(args) != 1 {
+			panic("config: bad arguments")
+		}
+
+		key := args[0].([]byte)
+
+		return getConfig(ctx, key)
+	case "SetConfig":
+		if len(args) != 3 {
+			panic("setConfig: bad arguments")
+		}
+
+		// check if it is inner ring invocation
+		irList := getInnerRingNodes(ctx, innerRingKey)
+		threshold := len(irList)/3*2 + 1
+
+		irKey := innerRingInvoker(irList)
+		if len(irKey) == 0 {
+			panic("setConfig: invoked by non inner ring node")
+		}
+
+		// check unique id of the operation
+		id := args[0].([]byte)
+		c := cheque{id: id}
+		cashedCheques := getCashedCheques(ctx)
+
+		chequesList, ok := addCheque(cashedCheques, c)
+		if !ok {
+			panic("setConfig: non unique id")
+		}
+
+		// vote for new configuration value
+		hashID := crypto.SHA256(id)
+
+		n := vote(ctx, hashID, irKey)
+		if n >= threshold {
+			removeVotes(ctx, hashID)
+
+			key := args[1]
+			val := args[2]
+
+			setConfig(ctx, key, val)
+			setSerialized(ctx, cashedChequesKey, chequesList)
+
+			runtime.Notify("SetConfig", id, key, val)
+			runtime.Log("setConfig: configuration has been updated")
+		}
+
+		return true
+	case "ListConfig":
+		var config []record
+
+		it := storage.Find(ctx, configPrefix)
+		for iterator.Next(it) {
+			key := iterator.Key(it).([]byte)
+			val := iterator.Value(it).([]byte)
+			r := record{key: key[6:], val: val}
```

Comment:
@AlexVanin, is `6` the length of `configPrefix`? If so, it would be convenient to clarify this through `len(configPrefix)` or contst.

Reply:
Yes, it is. I am agree with you, length will be more convenient there.